src/expandableCalendar/WeekCalendar/new.tsx

```diff
@@ -60,11 +60,13 @@ const WeekCalendar = (props: WeekCalendarProps) => {
 
   const onPageChange = useCallback(
     (pageIndex: number, _prevPage, {scrolledByUser}) => {
+      const dateData = items[pageIndex];
       if (scrolledByUser) {
-        context?.setDate(items[pageIndex], UpdateSources.WEEK_SCROLL);
+        context?.setDate(dateData, UpdateSources.WEEK_SCROLL);
       }
+      props.onDayPress?.(xdateToData(dateData));
```

Comment:
onDayPress is called when a day is pressed while onPageChange is called when there is a scroll on the list. We can not mix them